src/Features/CSharp/Portable/Completion/KeywordRecommenders/GlobalKeywordRecommender.cs

```diff
@@ -36,6 +36,7 @@ protected override bool IsValidContext(int position, CSharpSyntaxContext context
             return
                 context.IsStatementContext ||
                 context.IsGlobalStatementContext ||
+                UsingKeywordRecommender.IsUsingDirectiveContext(context, forGlobalKeyword: true, cancellationToken) ||
```

Comment:
i don't love this in that it seems to go against the general 'context' pattern we have.  but i'm not opposed to this, and i'm willing to make the change to unify this with the existing pattern if you're ok with that @AlekseyTs

Reply:
> and i'm willing to make the change to unify this with the existing pattern if you're ok with that 

Are you talking about unifing this in a follow-up PR, or in contect of this PR. I am not sure it is clear to me what you are offering.

---
In reply to: [599782529](https://github.com/dotnet/roslyn/pull/52059#discussion_r599782529) [](ancestors = 599782529)

Reply:
So, my preference is that we have a new boolean property on the context object (Called something like IsUsingDirectiveContext and IsGlobalUsingContext).  And then this recommender would just check that.  

I'm also saying that i'm not expecting that in this PR.  I'm also saying, if you'd rather i make that change, i'm happy to do that as well in a followup PR.  I don't expect you to have to make that change.  Thanks!

Reply:
>So, my preference is that we have a new boolean property on the context object (Called something like IsUsingDirectiveContext and IsGlobalUsingContext).  And then this recommender would just check that.  

> I'm also saying that i'm not expecting that in this PR.  I'm also saying, if you'd rather i make that change, i'm happy to do that as well in a followup PR.  I don't expect you to have to make that change.  Thanks!

Thanks for clarifying. I'll let you follow up on thit, especially that IsUsingDirectiveContext isn't a property on the context today and I am not sure if it is worth making it a property and precomputing its value (it looks like some properties on the context are precomputed) for scenarios when it is not even needed. 

---
In reply to: [599802941](https://github.com/dotnet/roslyn/pull/52059#discussion_r599802941) [](ancestors = 599802941)

Reply:
sounds good to me